acm_batch_invoice/models/batch_invoice_line.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 Ecosoft Co., Ltd (https://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import models, fields, api
+
+
+class BacthInvoiceLine(models.Model):
+    _name = 'batch.invoice.line'
+    _description = 'something'
+
+    batch_invoice_id = fields.Many2one(
+        comodel_name='batch.invoice',
+    )
+    agreement_id = fields.Many2one(
+        comodel_name='agreement',
+        string='Agreements'
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Partner',
+    )
+    lock = fields.Text()
+    water_amount = fields.Float()
+    water_from = fields.Float(
+        string='Water From',
+    )
+    water_to = fields.Float(
+        string='Water To',
+    )
+    electric_amount = fields.Float()
+    electric_from = fields.Float(
+        string='Electric From',
+    )
+    electric_to = fields.Float(
+        string='Electric To',
+    )
+
+    @api.onchange('water_to', 'water_from', 'electric_to', 'electric_from')
+    def _onchange_amount(self):
+        condition1 = self.water_to - self.water_from
```

Comment:
Please use meaningful variable, i.e.,
```suggestion
        water_diff = self.water_to - self.water_from
```